pkg/registry/core/service/rest.go

```diff
@@ -586,6 +534,54 @@ func (rs *REST) initClusterIP(service *api.Service) (bool, error) {
 	return false, nil
 }
 
+func (rs *REST) initNodePort(service *api.Service, nodePortOp *portallocator.PortAllocationOperation) error {
+	assignNodePorts := shouldAssignNodePorts(service)
```

Comment:
>Plus, is there easy way to extract part of the updateNodePort with initNodePort? If there is no straightforward way to do it, it is okay.

Didn't find an easy way to extract some common parts yet. It might be a future work.